test/functional/tool_wallet.py

```diff
@@ -46,9 +70,23 @@ def run_test(self):
         self.assert_raises_tool_error('Error loading wallet.dat. Is wallet being used by other process?', '-wallet=wallet.dat', 'info')
         self.assert_raises_tool_error('Error: no wallet file at nonexistent.dat', '-wallet=nonexistent.dat', 'info')
 
-        # stop the node to close the wallet to call info command
+        # Stop the node to close the wallet to call the info command.
+        self.log.info('Stop node\n')
         self.stop_node(0)
 
+        # Wallet tool info should work with wallet file permissions set to read-only.
+        # This needs to be fixed.
+        self.log.info('Set wallet file permissions to read-only')
+        os.chmod(self.wallet_path(), 0o400)
+        self.log.info('Call wallet tool info')
+        self.assert_raises_tool_error('Error loading . Is wallet being used by another process?', 'info')
+        self.log.info('Wallet tool info command raises loading error! **FIX ME**')
```

Comment:
TODO's shouldn't be logged, but added as inline comments

Reply:
Done. Waiting for Travis CI to pass locally and then will push a rebased update.